### PR TITLE
Fix object permissions enforcement on Job Buttons

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #4988 (will also be needed in 1.6.x)
# What's Changed

- The `job_buttons` template tag wasn't considering object-level permissions when determining whether to enable or disable each JobButton it rendered. I added the appropriate `has_perms` check.
  - There was a lot of duplicate code and logic in this function between the "ungrouped buttons" and "grouped buttons" cases, so I've extracted it into a helper function to reduce duplication.
- Similarly, the `JobButtonRunView` implementation was also not considering object-level permissions. In this case this view is redundant as a JobButton's underlying Job can be invoked just as easily through the `/extras/jobs/<id>/run/` view, which is already enforcing permissions correctly. I've therefore *removed `JobButtonRunView`* rather than attempt to fix it.
   - To keep the existing behavior where clicking a JobButton redirects back to the current page rather than redirecting to the JobResult page, I've added support for a `_return_url` parameter to the Job run view.
- Enforcing `extras.run_jobbutton` permission on top of `extras.run_job` permission was redundant, especially since a user with `extras.run_job` permission alone could still run the underlying job without going through the JobButton. I've therefore *removed `extras.run_jobbutton` as a required permission*; `extras.run_job` suffices.
- In manual testing, I found that the example plugin `ExampleComplexJobButtonReceiver` hadn't been completely/correctly updated to work under 2.0, so I fixed it.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- n/a Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/templatetags/job_buttons.py [hunk 1]
```diff
@@ -69,115 +70,108 @@
 </div>
 """
 
+SAFE_EMPTY = mark_safe("")  # noqa: S308
```
Comment (gsnider2195) [inline nautobot/extras/templatetags/job_buttons.py:73]:
```suggestion
SAFE_EMPTY_STR = mark_safe("")  # noqa: S308
```
Make it easier to determine what this variable is doing without context later in the file.

File: nautobot/extras/views.py [hunk 2]
```diff
@@ -1223,6 +1226,19 @@ def post(self, request, class_path=None, pk=None):
                     **job_model.job_class.serialize_data(job_kwargs),
                 )
 
+                return_url = request.POST.get("_return_url")
```
Comment (gsnider2195) [inline nautobot/extras/views.py:1229]: There's a change in behavior here that we'll want to make sure is desired. Previously job buttons ignored any of the error cases above this `else` block and were always redirected back to the object detail page.

**Review by timizuoebideri1 — APPROVED** [review]

File: nautobot/extras/views.py (hunk 2 above)
Comment (glennmatthews) [reply nautobot/extras/views.py:1229]: Good catch. Let me see if there's a straightforward way I can make the error cases also respect `_return_url`.

**Review by gsnider2195 — APPROVED** [review]
💯

**Pull request merged**